src/android/com/adobe/phonegap/push/PushPlugin.java

```diff
@@ -179,6 +181,20 @@ public void run() {
                     }
                 }
             });
+        } else if (SET_APPLICATION_ICON_BADGE_NUMBER.equals(action)) {
```

Comment:
If I understand correctly, this code only runs when the app is opened, not when the push message arrives (That's what happens on my Device, anyway). If this is indeed the case, it can't update the counter badge until the app opened, which doesn't solve the bug - no?